_posts/en/newsletters/2021-01-13-newsletter.md

```diff
@@ -0,0 +1,217 @@
+  can sign the dual funding transaction, they need the identities
+  (outpoints) of all of the UTXOs the other party wants to add to the
+  transaction.  This creates a risk that an abuser will attempt to
+  initiate dual-funded channels with many different users, learn their
+  UTXOs, and then refuse to sign the funding transaction---harming those
+  users' privacy at no cost to the abuser.
+
+    This week, Lloyd Fournier [posted][fournier podle] to the Lightning-Dev
+    mailing list an evaluation of two previous proposals to deal
+    with this problem, [one][zmn podle] using Proofs of Discrete Log
+    Equivalency (PoDLEs, see [Newsletter #83][news83 podle]) and the
+    [other][darosior sighash] using dual funding transactions
+    half-signed with `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY`.  Fournier
+    extended the previous half-signed proposal and then provided his own
+    proposal that was equivalently effective but simpler.  The new
+    proposal has the initiator create and sign (but not broadcast) a
+    transaction that spends their UTXO back to themselves.  They give
+    this to the other party as a proof of good faith---if the initiator
+    later fails to sign the actual funding transaction, the respondent
+    can broadcast the good-faith transaction, forcing the initiator to
+    pay an onchain fee.  Fournier concludes his post with a summary of
+    the tradeoffs between the different approaches.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Add unit testing of node eviction logic][review club
+#20477] is a PR ([#20477][Bitcoin Core #20477]) by
+pseudonymous contributor practicalswift to improve test coverage of Bitcoin Core's peer
+eviction logic when a node's inbound connection slots are full. Extreme care
+must be taken to avoid exposing the node to attacker-triggered network
+partitioning with this logic, so this code is critical.
+
+Most of the discussion focused on understanding Bitcoin Core's peer eviction
+logic.
+
+{% include functions/details-list.md
+  q0="Inbound or outbound peer selection: which one is our primary defense
+      against eclipse attacks?"
+  a0="Outbound peer selection, as outbound peers are far less under attacker
+      control than inbound connections. Inbound peer eviction is a second-order
+      protection---and not all nodes permit incoming connections."
+  a0link="https://bitcoincore.reviews/20477#l-77"
+
+  q1="Why does Bitcoin Core evict inbound connections?"
+  a1="To make inbound slots available for honest peers in the network so that
+      new nodes can establish good outbound connections to them. Otherwise,
+      dishonest nodes could more easily attack new nodes by being available for
+      their outbound connections and by statically occupying inbound slots."
```

Comment:
What does 'statically occupying inbound slots' mean? Do you mean occupy all the inbound slots of other listening nodes?

Reply:
Yes. I was trying to convey the idea of a more static network topology in the absence of￼ evictions.